### PR TITLE
Rubocop config adjust RE: Layout/LineLength

### PR DESCRIPTION
**Pull request by mjankowski**
As noted here - https://github.com/mastodon/mastodon/pull/26002#issuecomment-1643979994 - there's an upstream bug in rubocop todo generator now where certain configuration interactions around the `Layout/LineLength` setting wind up generating duplicate entries in the todo file. In our case because we also have the yaml lint running, this means there's a manual cleanup step involved in regenerating that file.

This PR starts by just regenerating the todo file (there were some changes from recent commits).

Then I added a high-ish `Max` config value for linelength. Then I stepped through one reduction at a time, dropping the limit a bit and fixing things.

I landed on 320 because when I tried to drop down much further there were so many violations that it just bailed on listing them and excluded the rule -- and I didn't want to do too many more corrections on what started as a configuration fix.

Basically all the changes here are (in my opinion) also nice renaming/readability refactors, so that's a nice side effect of the line length reductions as well.

This repeats some of https://github.com/mastodon/mastodon/pull/24174 but does not go as far in reducing the limit, and I tried to avoid adding `rubocop:disable` where possible and actually fix the violations.

Let me know if you'd prefer these be submitted as more explicit readability/naming refactors instead (which regenerate line limit as side effect) -- there's basically 1-2 files changing in each commit here and it would be easy to convert to that approach in individual PRs.

After the config changes the goal of "you can regenerate a new todo file and dont need to manually edit it to remove duplicate LineLength entries" appears to be achievd.

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -365,43 +364,6 @@ Rails/ApplicationController:
   Exclude:
     - 'app/controllers/health_controller.rb'
 
-# Configuration parameters: Database, Include.
-# SupportedDatabases: mysql, postgresql
-# Include: db/**/*.rb
-Rails/BulkChangeTable:
```
Comment (mjankowski) [inline .rubocop_todo.yml:371]: I believe this whole block is no longer considered eligible for review as side effect of the read replica config changes. Part of the cop is detecting a database key in the yaml which no longer exists. Arguably a bug in the cop, but I think we are not going to fix old migrations anyway.

File: app/models/concerns/account_followable.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module AccountFollowable
+  extend ActiveSupport::Concern
+
+  included do
+    scope :followable_by, ->(account) { without_follows(account).without_follow_requests(account) }
```
Comment (mjankowski) [inline app/models/concerns/account_followable.rb:7]: As far as I can tell, the SQL generated here did not change from previous.

File: config/initializers/twitter_regex.rb [hunk 3]
```diff
@@ -26,7 +26,9 @@ class Regex
         )
       \)
     /iox
+    # rubocop:disable Layout/LineLength
     UCHARS = '\u{A0}-\u{D7FF}\u{F900}-\u{FDCF}\u{FDF0}-\u{FFEF}\u{10000}-\u{1FFFD}\u{20000}-\u{2FFFD}\u{30000}-\u{3FFFD}\u{40000}-\u{4FFFD}\u{50000}-\u{5FFFD}\u{60000}-\u{6FFFD}\u{70000}-\u{7FFFD}\u{80000}-\u{8FFFD}\u{90000}-\u{9FFFD}\u{A0000}-\u{AFFFD}\u{B0000}-\u{BFFFD}\u{C0000}-\u{CFFFD}\u{D0000}-\u{DFFFD}\u{E1000}-\u{EFFFD}\u{E000}-\u{F8FF}\u{F0000}-\u{FFFFD}\u{100000}-\u{10FFFD}'
+    # rubocop:enable Layout/LineLength
```
Comment (mjankowski) [inline config/initializers/twitter_regex.rb:31]: As a side note -- is our intention with this file to basically "vendor" portions of this? https://github.com/twitter/twitter-text/blob/master/rb/lib/twitter-text/regex.rb

If so, some of it has changed a bit from what we have here.

File: .rubocop.yml [hunk 4]
```diff
@@ -38,14 +38,10 @@ Layout/FirstHashElementIndentation:
 # Reason: Currently disabled in .rubocop_todo.yml
 # https://docs.rubocop.org/rubocop/cops_layout.html#layoutlinelength
 Layout/LineLength:
-  AllowedPatterns:
-    # Allow comments to be long lines
-    - !ruby/regexp / \# .*$/
```
[the remark this answers is not in the document]
Comment (renchap) [reply .rubocop.yml:43]: Do we really want comments longer than 320 char? 🤔

[the remark this answers is not in the document]
Comment (mjankowski) [reply .rubocop.yml:43]: Maybe not! - but I was trying to not change more things than I had to.

**Comment by nschonni** [comment]
This will close #24174 too

**Comment by mjankowski** [comment]
I'm going to close this in favor of doing one PR per substantive change, rather than just churning through the line length reductions. If sufficient of those are accepted I'll submit a todo-regeneration-only PR for the length config.

**Pull request closed without merging**